library/ssl_tls13_client.c

```diff
@@ -1087,33 +1124,33 @@ int mbedtls_ssl_write_pre_shared_key_ext( mbedtls_ssl_context *ssl,
         *p++ = (unsigned char)( ext_length & 0xFF );
 
         /* 2 bytes length field for array of PskIdentity */
-        *p++ = (unsigned char)( ( ( ssl->conf->psk_identity_len + 4 + 2 ) >> 8 ) & 0xFF );
-        *p++ = (unsigned char)( ( ssl->conf->psk_identity_len + 4 + 2 ) & 0xFF );
+        *p++ = (unsigned char)( ( ( psk_identity_len + 4 + 2 ) >> 8 ) & 0xFF );
+        *p++ = (unsigned char)( ( psk_identity_len + 4 + 2 ) & 0xFF );
 
         /* 2 bytes length field for psk_identity */
-        *p++ = (unsigned char)( ( ( ssl->conf->psk_identity_len ) >> 8 ) & 0xFF );
-        *p++ = (unsigned char)( ( ssl->conf->psk_identity_len ) & 0xFF );
+        *p++ = (unsigned char)( ( ( psk_identity_len ) >> 8 ) & 0xFF );
+        *p++ = (unsigned char)( ( psk_identity_len ) & 0xFF );
 
         /* actual psk_identity */
-        memcpy( p, ssl->conf->psk_identity, ssl->conf->psk_identity_len );
+        memcpy( p, psk_identity, psk_identity_len );
 
-        p += ssl->conf->psk_identity_len;
+        p += psk_identity_len;
 
         /* Calculate obfuscated_ticket_age */
         /* ( but not for externally configured PSKs ) */
-        if( ssl->conf->ticket_age_add > 0 )
+        if( ssl->session_negotiate->ticket_age_add > 0 )
         {
 #if defined(MBEDTLS_HAVE_TIME)
             now = time( NULL );
 
-            if( !( ssl->conf->ticket_received <= now && now - ssl->conf->ticket_received < 7 * 86400 * 1000 ) )
+            if( !( ssl->session_negotiate->ticket_received <= now && now - ssl->session_negotiate->ticket_received < 7 * 86400 * 1000 ) )
```

Comment:
This is not related to this PR. But sounds like the `now` is in second, but the `7 * 86400 * 1000` is in millisecond. This means the ticket won't be expired after 1000 weeks.